### PR TITLE
Move token checking process to server-side

### PR DESCRIPTION
**Pull request by craigp-atw**
Moved tokenize and checking process to the server-side. Converted to an API call that returns an array/structure of tokens.
image

Added 'Check' button to prevent constant API requests.
image

### CODE REVIEW

File: app/src/lib/parser/index.js
Comment (longrunningprocess) [inline app/src/lib/parser/index.js:1]: you can just support the `async` in `pipe` where it's actually `await`-ed or not is up to the consumer, that way you only need one `pipe`

File: app/src/lib/pipeline.js
Comment (longrunningprocess) [inline app/src/lib/pipeline.js:1]: just as an experiment, try only using `pipe_async` and see if everything still so works the same, if so I'd go with a `pipe` that supports `asnyc/await`

File: app/src/lib/rules/case_frame/verbs.js [hunk 1]
```diff
@@ -382,9 +381,6 @@ const VERB_CASE_FRAME_RULES = create_verb_argument_rules()
  */
 export function check_verb_case_frames(tokens, verb_index, rules_modifier=rules => rules) {
 	const verb_token = tokens[verb_index]
-	if (!token_has_concept(verb_token)) {
```
Comment (longrunningprocess) [inline app/src/lib/rules/case_frame/verbs.js:385]: a deleted `if` is my fave 🏆

**Review by longrunningprocess — COMMENTED** [review]
double-check that feedback on the `pipe` and `async pipe` and see if it can be simplified or not.  Otherwise, I think this is ready to go.

**Comment by longrunningprocess** [comment]
after further thought and a little research, I guess the trade-off on the `async pipe` is to either have two separate calls or a condition inside the one that will determine whether it received a `Promise` or not.

**Review by longrunningprocess — APPROVED** [review]

**Comment by craigp-atw** [comment]
> after further thought and a little research, I guess the trade-off on the `async pipe` is to either have two separate calls or a condition inside the one that will determine whether it received a `Promise` or not. If you don't foresee a situation where the sequence of an async call and sync call don't conflict, then this solution will work but putting the condition inside `pipe` will help avoid such a conflict should it arise in the future.

I had tried only using the async pipe, but it caused a function in case_frame to fail (see below), because it returned a Promise instead of the expected value. I could have changed that function to become `async`, but that would have an unnecessary ripple effect throughout the whole rule system. 
image

**Pull request merged**